app/views/AdminPanelView/index.js

```diff
@@ -32,6 +33,7 @@ export default class AdminPanelView extends React.Component {
 		return (
 			<SafeAreaView style={styles.container} testID='terms-view'>
 				<StatusBar />
+				<NotificationBadge />
```

Comment:
It's working nicely, but you're repeating your self too much in this line.
You could do like this: https://reactnavigation.org/docs/en/custom-navigators.html#extending-navigators

Example overwriting `InsideStackModal` on `app/index`:
```jsx
class CustomNavigator extends React.Component {
	static router = InsideStackModal.router;

	render() {
		const { navigation } = this.props;
		return (
			<React.Fragment>
				<InsideStackModal navigation={navigation} />
				<NotificationBadge navigation={navigation} />
			</React.Fragment>
		);
	}
}
```

You'll have to change your notification style a little, because it's going to open from the top of the app instead of the bottom of the header.

Reply:
@diegolmello 
isTransitioning:false
index:0
key:InsideStack
routeName:InsideStack
params:undefined

this is what navigation.state is returning see the route name is always InsideStack , there is no way of knowing which view I am If I this logic , and because of that I cannot check for condition in which notification is not shown if your is inside the same room .. also it create difficulty in navigating ...